### PR TITLE
Add nimblenote.app v3.1.1

### PR DESCRIPTION
**Pull request by davidwinter**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by davidwinter** [comment]
Appears as though this has failed as it is assuming that the GitHub repo is where the source is located, whereas it is only used for downloads and website source. Is there a way to workaround this?

File: Casks/nimblenote.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  url "https://github.com/nimblenote/nimblenote/releases/latest/download/nimblenote-#{version}.dmg", verified: "https://github.com/nimblenote/nimblenote"
+  name "nimblenote"
+  desc "Keyboard-driven note taking"
+  homepage "https://nimblenote.app/"
+
+  livecheck do
+    url "https://github.com/nimblenote/nimblenote/releases/latest/download/release.json"
+    strategy :page_match
+    regex(/"version":"(\d+\.\d+\.\d+)"/i)
+  end
```
Comment (bevanjkay) [inline Casks/nimblenote.rb:14]: @davidwinter The livecheck stanza is not required when the download URL is github, as it uses the `:git` strategy by default.

**Review by bevanjkay — CHANGES_REQUESTED** [review]

**Comment by bevanjkay** [comment]
> From what I can tell, if the app wasn't using GitHub for release downloads, then this check wouldn't even be happening and the submission would have passed the checks?

Generally, the notability threshold can be disregarded when it is only the binaries that are hosted on Github.

Additionally, can you please try to add a [zap stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

File: Casks/nimblenote.rb (hunk 1 above)
Comment (davidwinter) [reply Casks/nimblenote.rb]: Does that integrate with GitHub releases, as the main application source code does not actually live within that git repository?

**Comment by davidwinter** [comment]
@bevanjkay thanks for the clarification - I've also added the `zap` stanza. I've purposefully not removed the entire `Application Support/nimblenote` directory, as some user-created files exist underneath it that we wouldn't want to delete.

**Comment by bevanjkay** [comment]
> @bevanjkay thanks for the clarification - I've also added the `zap` stanza. I've purposefully not removed the entire `Application Support/nimblenote` directory, as some user-created files exist underneath it that we wouldn't want to delete.

What kind of user created files? The zap stanza only runs if it is invoked with the --zap switch on `brew uninstall`

**Comment by davidwinter** [comment]
> > @bevanjkay thanks for the clarification - I've also added the `zap` stanza. I've purposefully not removed the entire `Application Support/nimblenote` directory, as some user-created files exist underneath it that we wouldn't want to delete.
> 
> What kind of user created files? The zap stanza only runs if it is invoked with the `--zap` switch on `brew uninstall`

User notes created by the application are stored within that directory by default, so not something to trash.

**Comment by davidwinter** [comment]
@bevanjkay I _think_ I've done all the extra review points and checks I can now with this. It's still failing on the GitHub checks, but I'm assuming that is something that can be handled when it comes to merging?

**Comment by miccal** [comment]
Thank you @davidwinter.

**Comment by davidwinter** [comment]
Thanks @miccal - I had a few questions I just wanted to check up on which I'm then happy to open another PR if we decide appropriate:

- the `rmdir` that you added to the `zap` stanza; will that only remove the directory if it is empty? Otherwise, I didn't add this because any notes that a user creates with the application are stored within this directory, and I fear that adding this directory to be removed, rather than specific files and directories underneath it, will remove the user-created notes as well
- the name of the app is `nimblenote` rather than `Nimblenote`

**Pull request merged**